_posts/2019-11-28-making-software-is-hard.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: Making Software Is Hard
+date: 2019-11-28 16:28:05.000000000 +01:00
+type: post
+published: true
+status: publish
+categories: [Programming]
+image:
+image2:
+author: Valentino Urbano
+---
+
+It is hard for non technical people to understand the amount of work that goes on behind the scene in creating any kind of non trivial application. It is even harder to make them understand that 'just changing x' might not be as trivial as they think it would be. Only as a software engineer yourself[^1] you can assume with some degree of confidence what would generally be the amount of work required for a certain feature.
+
+I think that we have finally reached a point where everybody pretty much understands that making software is not easy. The only people who don't are junior programmers (no pun intended, we all were juniors once) or people who take an already made application and just change the visuals (themes, re-skins,..) and call themselves software engineers. What we have not realized is the magnitude on how much it is not.
+
+I doubt we'll reach a point where the general population understands that most changes they think about are actually much harder to implement than they imagine anytime soon (if ever).
+
+Sure, you might argue that some times it is the opposite. Changing something may just mean changing one line of code when from an outside perspective it might seem like not such a trivial change, but the opposite case is 99% more common.
+
+## Estimates
+
+### Deadlines
+
+Estimates in software are notoriously hard. People underestimate work by an order of magnitude. Whatever the metodology to do estimates that we use we usually think about the new feature to develop in isolation, but we don't estimate the time that it might take for hiccups, interfacing with the rest of the system, having other priorities that limit our available time to work on such feature and so on. This results in an industry that in general keeps failing to keep its promise given at estimation time with projects constantly over budget and delayed. Public projects are notorious for this (healtcare.gov as a "fairly recent" example that got a lot of media coverage), but in private projects the same is true as well.
```

Comment:
metodology -> methodology

Reply:
And that sentence needs rewriting...

Reply:
estimation - > estimated

Reply:
> estimation - > estimated

given at estimation time

Don't think that ed is correct here